contracts/src/chainlink/solidity/emergency/StarkNetValidator.sol

```diff
@@ -82,7 +100,18 @@ contract StarkNetValidator is TypeAndVersionInterface, AggregatorValidatorInterf
     payload[1] = block.timestamp;
     // Make the StarkNet x-domain call.
     // NOTICE: we ignore the output of this call (msgHash, nonce), and we don't raise any events as the event LogMessageToL2 will be emitted from the messaging contract
-    STARKNET_CROSS_DOMAIN_MESSENGER.sendMessageToL2(L2_UPTIME_FEED_ADDR, STARK_SELECTOR_UPDATE_STATUS, payload);
+    STARKNET_CROSS_DOMAIN_MESSENGER.sendMessageToL2{value: s_fee}(
+      L2_UPTIME_FEED_ADDR,
+      STARK_SELECTOR_UPDATE_STATUS,
+      payload
+    );
     return true;
   }
+
+  /**
+   * @notice makes this contract payable
+   * @dev receives funds:
+   *  - funds are used to send cross chain messages to L2
+   */
+  receive() external payable {}
```

Comment:
We also need the ability for the owner to withdraw funds from this contract, for example when we migrate to a different contract for whatever reason.

Also we need to investigate how message cancelations work, and do we require any additional logic in this contract except the `receive` fn to be refunded: https://docs.starknet.io/docs/L1-L2%20Communication/messaging-mechanism#l1--l2-message-fees

> Throughtout the lifecycle of the message, the assoicated fee is locked in the Core contract. If the message was not processed for any reason, then [cancelling](https://docs.starknet.io/docs/L1-L2%20Communication/messaging-mechanism#l1--l2-message-cancellation) the message will also return the fee to the sender.

Reply:
I've added the withdrawal functionality.

We still need to investigate message cancelations.

Reply:
We can cancel messages by calling the `cancelL1ToL2Message` on the `Starknet` contract.

https://goerli.etherscan.io/address/0xde29d060D45901Fb19ED6C6e959EB22d8626708e#writeProxyContract.

The tricky thing about this is that we'd have to store the nonce and status in our contract before we send the cross chain message.  This might get expensive.

Reply:
If we ever need to do that we can read that data from the raised bridge events, and invoke it manually.

I assume that invocation will refund the funds to the contract, that's the only thing that makes sense to do.